src/ItemManager.js

```diff
@@ -0,0 +1,52 @@
+
+class ItemManager {
+    constructor() {
+        this.tasks = [];
+        this.pokemonClient = new PokemonClient();
+    }
+
+    async handleTask(task) {
+        let taskSplit = task.split(",");
+        if (isNaN(taskSplit[0])) {
+            return [[task], false];
+        }
+        if (taskSplit.length === 1){
+            // fetch 1 pokemon
+            const [pokemon, success] = await this.pokemonClient.getPokemon(taskSplit[0]);
+            if (!success){
+                return [[pokemon], true];
+            }
+            const taskName = `Catch ${pokemon.name}, the ${pokemon.types[0].type.name} type pokemon`;
+            return [[taskName], true];
+        }
+        // handle list of pokemons
+        const result = await this.pokemonClient.getListPokemon(taskSplit);
+        if (!result[1]) {
+            return [[result[0]], true]
+        }
+        let pokemons = [];
+        for (let p of result[0]) {
+            const pokemon = await p;
+            const taskName = `Catch ${await pokemon.name}, the ${pokemon.types[0].type.name} type pokemon`;
```

Comment:
this line is duplicated i would extract that to a private function here in the class